Ensure sensor_channels not local channels are used when subsetting

CoryMartin-NOAA opened the pull request.

ADCollard committed:
Ensure sensor_channels not local channels are used when subsetting

CoryMartin-NOAA commented:
work done by @ADCollard

kevindougherty-noaa commented:
Looks good once you fix code norms!

ADCollard committed:
Update ioda_obs_space.py

danholdaway commented on src/eva/data/ioda_obs_space.py:
```diff
@@ -97,6 +97,9 @@ def execute(self, data_collections):
                 ds_groups = xr.Dataset()
 
+                # Save sensor_channels for later
+                sensor_channels = ds_header['nchans']
```
I think this code might need to be protected to ensure it's not used when `nchans` is not in the file.

CoryMartin-NOAA replied:
Yeah so then the first one would need:
```
has_sensor_channels = False
if 'nchans' in ds_header.keys():
    sensor_channels = ds_header['nchans']
    has_sensor_channels = True
```
and then the later line:

```
if has_sensor_channels:
    ds['nchans'] = sensor_channels
```

Or some other similar solution

danholdaway replied:
And that works too.

danholdaway reviewed (changes requested):
One comment, that I think is why there's a failure of the application test.

ADCollard commented:
@CoryMartin-NOAA, the coding norms are fixed but for some reason the eva application test is failing for aircraft.  So I will investigate further.

danholdaway commented on src/eva/data/ioda_obs_space.py:
```diff
@@ -97,6 +97,9 @@ def execute(self, data_collections):
                 ds_groups = xr.Dataset()
 
+                # Save sensor_channels for later
+                sensor_channels = ds_header['nchans']
```
```suggestion
sensor_channels = 0
if 'nchans' in ds_header.keys():
    sensor_channels = ds_header['nchans']
```

ADCollard committed:
Update src/eva/data/ioda_obs_space.py

Was about to commit the same thing!  Thanks!

Co-authored-by: Dan Holdaway <27729500+danholdaway@users.noreply.github.com>

CoryMartin-NOAA commented on src/eva/data/ioda_obs_space.py:
```diff
@@ -98,7 +98,9 @@ def execute(self, data_collections):
 
                 # Save sensor_channels for later
-                sensor_channels = ds_header['nchans']
+sensor_channels = 0
```
@ADCollard great indentation! :-)

ADCollard replied:
@CoryMartin-NOAA Not sure what happened there, I accepted some suggested changes and it ended up like that.

ADCollard committed:
Update ioda_obs_space.py

I accepted suggested changes earlier but they came in corrupted.   So hopefully this fixes it!

CoryMartin-NOAA commented on src/eva/data/ioda_obs_space.py:
```diff
@@ -139,6 +144,10 @@ def execute(self, data_collections):
                     ds = ds.rename(rename_dict)
 
+                    # Reset channel numbers from header
+                    if 'nchans' in ds.keys()
```
I'm not sure this will work, have you tested it? Because `ds.keys()` is for the group which may not contain the header information? But I'm not sure.

ADCollard committed:
Update ioda_obs_space.py

Hopefully finally fixing the nchans exception
